File: coursework_console/MenuComponents/FileInfo.cpp
#include <fstream>
#include <nlohmann/json.hpp>
#include "FileInfo.h"
#include "OrdersData.h"

using json = nlohmann::json;

std::string FileInfo::getPath() {
	if (path == "") {
		std::ifstream properties("..\\properties.json");
		json propertiesJson = json::parse(properties);
		properties.close();
		setPath(propertiesJson["lastFile"]);
	}
	return path;
}

void FileInfo::setPath(const std::string& newPath) {
	path = newPath;
	OrdersData::loadOrders(path);
}

bool FileInfo::isPathSet() {
	return getPath() != "";
}

File: courseworkWinFormsCSharp/DataManagers/FileInfo.cs
using Newtonsoft.Json.Linq;
using System.IO;

namespace courseworkWinFormsCSharp.DataManagers
{
    internal static class FileInfo
    {
        private static string path = null;
        public static string Path
        {
            get
            {
                if (string.IsNullOrEmpty(path))
                {
                    using (StreamReader properties = new StreamReader("..\\properties.json"))
                    {
                        path = JObject.Parse(properties.ReadToEnd()).GetValue("lastFile").ToString();
                    }
                }
                return path;
            }
            set
            {
                path = value;
                OrdersData.LoadOrders(path);
            }
        }
    }
}
